ffi/derive/src/derive.rs

```diff
@@ -203,22 +175,48 @@ fn gen_ffi_fn_body(method_name: &Ident, field_ty: &Type, derive: Derive) -> Toke
 fn gen_ffi_derive(struct_name: &Ident, field: &syn::Field, derive: Derive) -> syn::ItemFn {
     let field_name = field.ident.as_ref().expect_or_abort("Defined");
 
-    let mut field_ty = field.ty.clone();
-    if let Type::Path(field_ty) = &mut field_ty {
-        SelfResolver::new(&parse_quote! { #struct_name }).visit_type_path_mut(field_ty);
-    }
+    let self_ty = parse_quote! {#struct_name};
+    let field_ty = field.ty.clone();
+    let (handle, arg) = match derive {
+        Derive::Setter => (
+            Arg::handle(&self_ty, parse_quote! {&mut #struct_name}),
+            Arg::input(&self_ty, parse_quote! {field}, field_ty),
+        ),
+        Derive::Getter => (
+            Arg::handle(&self_ty, parse_quote! {&#struct_name}),
+            Arg::output(&self_ty, parse_quote! {output}, parse_quote! {&#field_ty}),
+        ),
+        Derive::MutGetter => (
+            Arg::handle(&self_ty, parse_quote! {&mut #struct_name}),
+            Arg::output(
+                &self_ty,
+                parse_quote! {output},
+                parse_quote! {&mut #field_ty},
+            ),
+        ),
+    };
 
     let derive_method_name = gen_derive_method_name(field_name, derive);
     let ffi_fn_name = gen_ffi_fn_name(struct_name, &derive_method_name);
     let ffi_fn_doc = gen_ffi_docs(struct_name, &derive_method_name);
-    let ffi_fn_args = gen_ffi_fn_args(struct_name, &field_ty, derive);
-    let ffi_fn_body = gen_ffi_fn_body(&derive_method_name, &field_ty, derive);
+    let ffi_fn_args = gen_ffi_fn_args(&handle, &arg, derive);
+    let ffi_fn_body = gen_ffi_fn_body(&derive_method_name, &handle, &arg, derive);
 
     parse_quote! {
         #[doc = #ffi_fn_doc]
         #[no_mangle]
         pub unsafe extern "C" fn #ffi_fn_name(#ffi_fn_args) -> iroha_ffi::FfiResult {
-            #ffi_fn_body
+            let res = std::panic::catch_unwind(|| {
```

Comment:
I'm not sure where a panic might hit, so I thought it's better to be safe then sorry

Reply:
Ideally we should have proper error handling and no panics.  #2252 is relevant.

Reply:
yeah, this is a form of precaution and I wouldn't remove it. It's hard to guarantee that no downstream library will ever panic